File: src/java/classes.java
public class classes {

    private static class MontyPython{
        private String _firstName;
        private String _lastName;
        public MontyPython(String firstName, String lastName){
            _firstName = firstName;
            _lastName = lastName;
        }

        public String FullName(){
            return _firstName + " " + _lastName;
        }

        public void set_FirstName(String firstName){
            _firstName = firstName;
        }

        public void set_LastName(String lastName){
            _lastName = lastName;
        }
    }
    public static void main(String[] args) {
        MontyPython montyPython = new MontyPython("John", "Cleese");
        System.out.println(String.format("Monty Python is %s.", montyPython.FullName()));

        montyPython.set_FirstName("Eric");
        montyPython.set_LastName("Idle");
        System.out.println(String.format("Now, Monty Python is %s.", montyPython.FullName()));
    }

}

File: src/scala/classes.scala
object Classes extends App {
    class MontyPython(private var _firstName: String, private var _lastName:String) {
        def set_firstName(firstName: String) = {
            _firstName = firstName
        }

        def set_lastName(lastName: String) = {
            _lastName = lastName
        }

        def fullName() :String = {
            return _firstName + " " + _lastName
        }
    }



    val montyPython = new MontyPython("John", "Cleese")
    println(s"Monty Phython is ${montyPython.fullName}")

    montyPython.set_firstName("Eric")
    montyPython.set_lastName("Idle")
    println(s"Now, Monty Python is ${montyPython.fullName}")
}